### ISSUE TITLE
None of the rendering classes are available for importing

### ISSUE THREAD

**Issue by mschleeweiss** [SAP/ui5-typescript#293]
Right now the following code cannot be implemented in TypeScript, because InputRenderer cannot be imported.

```js
sap.ui.define([
    "sap/m/InputRenderer"
], function(InputRenderer) {

    return InputRenderer.extend("de.sample.MyCustomInputRenderer", {

        writeInnerAttributes(oRm, oControl) {
            InputRenderer.writeInnerAttributes.apply(this, arguments);
            // my custom code
        }
    });

}, true)
```

In fact it seems like non of the Renderer classes can be imported. Is there a reason for this?

**Comment by akudev** [comment]
I can confirm the basic fact... I guess caused by the renderers not being public (they also don't appear in the API documentation). Most renderer stuff is not really considered public, but inheriting from renderers is indeed supposed to be possible...

**Comment by akudev** [comment]
Ok, let's get this forward. Overall, this is a tricky request, as opening the renderers for extension is considered "dangerous" by many (in the sense that it may cause error reports when something in the controls changes - the internals of a control's rendering are not considered stable API).
So I don't think we will see renderers generally going public.

The InputRenderer, however, is a bit special, as extending it is [even documented](https://sapui5.hana.ondemand.com/#/topic/bcee26a4801748f39bf5698d83d903aa) and it is probably the most likely control to be extended anyway. Hence I opened internal ticket 2280038175 to discuss with the control owners whether it can be made public.

Your code sample, however, will not work this way, as Renderers are no real classes in UI5 and have no extend() method on their own. Have a look at the [documentation](https://sapui5.hana.ondemand.com/#/topic/bcee26a4801748f39bf5698d83d903aa) to see it. When you inherit from Input, you automatically inherit the renderer (not FROM the renderer) and you can override single methods. You could probably also call Renderer.extend() like the controls do:
```js
var InputRenderer = Renderer.extend(InputBaseRenderer); // Renderer is sap/ui/core/Renderer !
```
But this returns "object" right now, so it's not super useful in TypeScript world.
All of this of course only once and if my proposal to make it public (or rather protected, which doesn't make a difference in our type definitions right now) is accepted.

**Comment by akudev** [comment]
Fixed by https://github.com/SAP/openui5/commit/d0d64fb026dfea00a67191c717b47fe68892d5ae for the InputRenderer. 

Other controls are right now not intended to be extended and aiming for renderer method stability. If you still dare to extend other renderers and adapt your code in case of changes, then you can do this by defining your own types mimicking the renderer ones.

Fix will be in 1.99

Thanks!
